tests/integration/test_tls.py

```diff
@@ -119,19 +136,9 @@ async def test_mattermost_db(ops_test: OpsTest) -> None:
         # Restart the initial primary and check the logs to ensure TLS is being used by pg_rewind.
         logger.info(f"starting database on {primary}")
         await run_command_on_unit(ops_test, primary, "/charm/bin/pebble start postgresql")
-        for attempt in Retrying(
-            stop=stop_after_delay(60 * 3), wait=wait_exponential(multiplier=1, min=2, max=30)
-        ):
+        for attempt in Retrying(stop=stop_after_delay(60 * 3), wait=wait_fixed(2), reraise=True):
             with attempt:
-                logger.info(f"checking if pg_rewind used TLS on {replica}")
-                logs = await run_command_on_unit(
-                    ops_test,
-                    replica,
-                    "grep rewind /var/log/postgresql/postgresql.log",
-                )
-                assert (
-                    "connection authorized: user=rewind database=postgres SSL enabled" in logs
-                ), "TLS is not being used on pg_rewind connections"
```

Comment:
This check consistently failed with secrets enabled. Reworked to check all units.

Reply:
I think we can check only the current leader and sync-stanby, in where the connections should be done.